### PR TITLE
[Perpetual] net open interest and funding rate functions were update.

### PR DESCRIPTION
**Pull request by fenriz07**

### CODE REVIEW

**Review by amityadav0 — COMMENTED** [review]
I think we also need to update this [here](https://github.com/elys-network/elys/blob/f066d72b196433ffca490366d1b0a6098312a52f/x/perpetual/keeper/settle_funding_fee_distribution.go#L19)

- Maybe we can reuse the current function, return 3 values
`assetLong`, `assetShort` and `net`
- Reuse first two values in `SettleFundingFeeDistribution`

File: x/perpetual/keeper/get_net_open_interest.go [hunk 1]
```diff
@@ -6,16 +6,68 @@ import (
...
+		} else {
+
+			coin := sdk.NewCoin(asset.AssetDenom, asset.Liabilities)
+			ammPool, err := k.GetAmmPool(ctx, pool.AmmPoolId, asset.AssetDenom)
+
+			if err != nil {
+				return sdk.ZeroInt()
+			}
+
+			l, err := k.OpenLongChecker.EstimateSwapGivenOut(ctx, coin, uusdc.Denom, ammPool)
```
Comment (amityadav0) [inline x/perpetual/keeper/get_net_open_interest.go:36]:
```suggestion
			estimatedSwap, err := k.OpenLongChecker.EstimateSwapGivenOut(ctx, coin, uusdc.Denom, ammPool)
```

File: x/perpetual/keeper/get_net_open_interest.go [hunk 2]
```diff
@@ -6,16 +6,68 @@ import (
...
+			if err != nil {
+				return sdk.ZeroInt()
+			}
+
+			assetLiabilitiesLong = assetLiabilitiesLong.Add(l)
```
Comment (amityadav0) [inline x/perpetual/keeper/get_net_open_interest.go:41]:
```suggestion
			assetLiabilitiesLong = assetLiabilitiesLong.Add(estimatedSwap)
```

File: x/perpetual/keeper/get_net_open_interest.go [hunk 3]
```diff
@@ -6,16 +6,68 @@ import (
...
+		} else {
+
+			coin := sdk.NewCoin(asset.AssetDenom, asset.Liabilities)
+			ammPool, err := k.GetAmmPool(ctx, pool.AmmPoolId, asset.AssetDenom)
+
+			if err != nil {
+				return sdk.ZeroInt()
+			}
+
+			l, err := k.OpenLongChecker.EstimateSwapGivenOut(ctx, coin, uusdc.Denom, ammPool)
```
Comment (amityadav0) [inline x/perpetual/keeper/get_net_open_interest.go:63]:
```suggestion
			estimatedSwap, err := k.OpenLongChecker.EstimateSwapGivenOut(ctx, coin, uusdc.Denom, ammPool)
```

File: x/perpetual/keeper/get_net_open_interest.go [hunk 4]
```diff
@@ -6,16 +6,68 @@ import (
...
+			if err != nil {
+				return sdk.ZeroInt()
+			}
+
+			assetLiabilitiesShort = assetLiabilitiesShort.Add(l)
```
Comment (amityadav0) [inline x/perpetual/keeper/get_net_open_interest.go:68]:
```suggestion
			assetLiabilitiesShort = assetLiabilitiesShort.Add(estimatedSwap)
```

File: x/perpetual/keeper/get_net_open_interest.go [hunk 5]
```diff
@@ -6,16 +6,68 @@ import (
...
+			}
+
+			estimatedSwap, err := k.OpenLongChecker.EstimateSwapGivenOut(ctx, coin, uusdc.Denom, ammPool)
+			if err != nil {
+				return sdk.ZeroInt()
+			}
+
+			assetLiabilitiesLong = assetLiabilitiesLong.Add(estimatedSwap)
+		}
+
```
Comment (cosmic-vagabond) [inline x/perpetual/keeper/get_net_open_interest.go:43]: @fenriz07 you seem to be repeating the same code at many places, could you create a separate function that is called instead, can you also apply those changes to here as @amityadav0 mentioned, thanks

Comment (cosmic-vagabond) [reply x/perpetual/keeper/get_net_open_interest.go]: @fenriz07 @amityadav0 updated PR with the changes mentioned above include `settle_funding_fee_distribution`

**Review by cosmic-vagabond — APPROVED** [review]

**Pull request merged**